receiver/syslogreceiver/syslog.go

```diff
@@ -62,19 +63,35 @@ func (f ReceiverType) BaseConfig(cfg config.Receiver) adapter.BaseConfig {
 
 // SysLogConfig defines configuration for the syslog receiver
 type SysLogConfig struct {
+	syslog.Config      `mapstructure:",squash"`
 	adapter.BaseConfig `mapstructure:",squash"`
-	Input              adapter.InputConfig `mapstructure:",remain"`
 }
 
 // DecodeInputConfig unmarshals the input operator
 func (f ReceiverType) DecodeInputConfig(cfg config.Receiver) (*operator.Config, error) {
 	logConfig := cfg.(*SysLogConfig)
-	yamlBytes, _ := yaml.Marshal(logConfig.Input)
-	inputCfg := syslog.NewConfig("syslog_input")
-	inputCfg.BaseConfig = syslogparser.NewConfig("syslog_parser").BaseConfig
+	return &operator.Config{Builder: &logConfig.Config}, nil
+}
+
+func (cfg *SysLogConfig) Unmarshal(componentParser *confmap.Conf) error {
+	if componentParser == nil {
+		// Nothing to do if there is no config given.
+		return nil
+	}
 
-	if err := yaml.Unmarshal(yamlBytes, &inputCfg); err != nil {
-		return nil, err
+	if componentParser.IsSet("tcp") {
+		cfg.TCP = CreateDefaultTCPConfigBase()
+	} else if componentParser.IsSet("udp") {
+		cfg.UDP = CreateDefaultUDPConfigBase()
 	}
-	return &operator.Config{Builder: inputCfg}, nil
+
+	return componentParser.UnmarshalExact(cfg)
+}
+
+func CreateDefaultTCPConfigBase() *tcp.BaseConfig {
```

Comment:
do you need this public?

Reply:
I just removed these functions, as they were only used once.

Reply:
Correction: These functions are used also for `internal/components/receiver_test.go`, which is why they are exported.

Reply:
I don't thing they should, we need to limit public API as much as possible.

Reply:
Especially when only for testing.

Reply:
I'll submit another PR shortly